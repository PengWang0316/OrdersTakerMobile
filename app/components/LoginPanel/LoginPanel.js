import React from 'react';
import { Text, View } from 'react-native';
import { Divider } from 'react-native-elements';

import LoginForm from '../LoginForm';
import SocialLoginPanel from '../SocialLoginPanel';
import styles from './Styles';

type Props = { handleSnackbarUpdate: Function, navigation: Object };

export const LoginPanel = ({ handleSnackbarUpdat, navigation }: Props) => (
  <View style={styles.internalContainer}>
    <SocialLoginPanel />

    <View style={styles.deviderView}>
      <Divider style={styles.devider} />
      <Text style={styles.title}>or</Text>
      <Divider style={styles.devider} />
    </View>

    <LoginForm handleSnackbarUpdat={handleSnackbarUpdat} navigation={navigation} />
  </View>
);
export default LoginPanel;
