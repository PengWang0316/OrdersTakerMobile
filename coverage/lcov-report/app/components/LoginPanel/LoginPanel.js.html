<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/LoginPanel/LoginPanel.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">app/components/LoginPanel</a> LoginPanel.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>15/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Text, View } from 'react-native';
import { Divider } from 'react-native-elements';
&nbsp;
import LoginForm from '../LoginForm';
import SocialLoginPanel from '../SocialLoginPanel';
import styles from './Styles';
&nbsp;
type Props = { handleSnackbarUpdate: Function };
&nbsp;
export const LoginPanel = ({ handleSnackbarUpdat }: Props) =&gt; (
  &lt;View style={styles.internalContainer}&gt;
    &lt;SocialLoginPanel /&gt;
&nbsp;
    &lt;View style={styles.deviderView}&gt;
      &lt;Divider style={styles.devider} /&gt;
      &lt;Text style={styles.title}&gt;or&lt;/Text&gt;
      &lt;Divider style={styles.devider} /&gt;
    &lt;/View&gt;
&nbsp;
    &lt;LoginForm handleSnackbarUpdat={handleSnackbarUpdat} /&gt;
  &lt;/View&gt;
);
export default LoginPanel;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 30 2018 14:48:02 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
