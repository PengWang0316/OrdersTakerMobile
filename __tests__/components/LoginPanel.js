import React from 'react';
import renderer from 'react-test-renderer';

import { LoginPanel } from '../../app/components/LoginPanel/LoginPanel';

// jest.mock('Text', () => 'Text');
// jest.mock('View', () => 'View');
// jest.mock('../../app/components/LoginForm', () => 'LoginForm');
// jest.mock('../../app/components/SocialLoginPanel', () => 'SocialLoginPanel');

describe('LoginPanel', () => {
  const defaultProps = { handleSnackbarUpdate: jest.fn() };
  test('Snapshot', () => expect(renderer.create(<LoginPanel {...defaultProps} />).toJSON()).toMatchSnapshot());
});
