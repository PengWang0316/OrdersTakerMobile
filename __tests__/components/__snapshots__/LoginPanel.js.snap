// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPanel Snapshot 1`] = `
<View
  style={
    Object {
      "width": "75%",
    }
  }
>
  <SocialLoginPanel />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 30,
        "marginTop": 30,
      }
    }
  >
    Divider
    <Text>
      or
    </Text>
    Divider
  </View>
  <LoginForm />
</View>
`;
